src/utils/build.ts

```diff
@@ -0,0 +1,143 @@
+
+import * as shellTools from './shell';
+
+export enum ClusterType {
+	hub, managed
+}
+
+export interface Cluster {
+	name: string,
+	context: string
+	type: ClusterType
+}
+
+export interface ProgressReport {
+	increment: number,
+	message: string
+}
+
+export const defaultClusters: Cluster[] = [
+	{
+		name: 'hub',
+		context: 'kind-hub',
+		type: ClusterType.hub
+	},
+	{
+		name: 'cluster1',
+		context: 'kind-cluster1',
+		type: ClusterType.managed
+	},
+	{
+		name: 'cluster2',
+		context: 'kind-cluster2',
+		type: ClusterType.managed
+	},
+];
+
+// create a kind cluster, fulfilled with stdout/stderr
+async function createKindCluster(cluster: Cluster):  Promise<string> {
+	console.debug(`creating a kind cluster named ${cluster.name}`);
+	return shellTools.executeShellCommand(`kind create cluster --name ${cluster.name}`);
```

Comment:
shell js does not return a Promise. 
how do you intended to run multiple request in parallel

Reply:
this is not _shelljs_.
this is our wrapper: _shellTools.executeShellCommand_ which returns a promise.

https://github.com/TomerFi/ocm-vscode-extension/blob/3-kind-dev-environment-dev/src/utils/shell.ts#L7